edsdme/blocks/announcements-preview/announcements-preview.js

```diff
@@ -0,0 +1,147 @@
+  };
+
+  const app = document.createElement('announcements-preview');
+
+  Array.from(el.children).forEach((row) => {
+    const cols = Array.from(row.children);
+    const rowTitle = cols[0].innerText.trim().toLowerCase().replace(/ /g, '-');
+
+    if (rowTitle && rowTitle === 'title') {
+      blockData.title = cols[1].innerText;
+    }
+    if (rowTitle && rowTitle === 'page-url') {
+      blockData.link = cols[1].innerText;
+    }
+    if (rowTitle && rowTitle === 'button-text') {
+      blockData.buttonText = cols[1].innerText;
+    }
+  });
+
+  function addAnnouncement(cardData) {
+    const linkWrapper = document.createElement('a');
+    linkWrapper.className = 'link-wrapper';
+    linkWrapper.href = cardData.contentArea.url;
+    linkWrapper.target = '_blank';
+
+    linkWrapper.style.display = 'block';
+
+    const announcementItem = document.createElement('div');
+    announcementItem.className = 'announcement-item';
+
+    const contentWrapper = document.createElement('div');
+    contentWrapper.className = 'card-content';
+    const titleElement = document.createElement('h3');
+    titleElement.className = 'card-title';
+    titleElement.textContent = cardData.contentArea.title;
+
+    const descriptionElement = document.createElement('p');
+    descriptionElement.className = 'card-description';
+    descriptionElement.textContent = cardData.contentArea.description;
+
+    const imageWrapper = document.createElement('div');
+    imageWrapper.className = 'card-image';
+
+    const picture = document.createElement('picture');
+
+    const source = document.createElement('source');
+    source.srcset = `${new URL(cardData.styles?.backgroundImage).pathname}?width=240&format=webp&optimize=small`;
+    source.type = 'image/webp';
+
+    const img = document.createElement('img');
+    img.src = `${new URL(cardData.styles?.backgroundImage).pathname}??width=240&format=webp&optimize=small`;
+    img.alt = 'Image description';
+    img.loading = 'lazy';
+
+    picture.appendChild(source);
+    picture.appendChild(img);
+
+    imageWrapper.appendChild(picture);
+    announcementItem.appendChild(imageWrapper);
+    contentWrapper.appendChild(titleElement);
+    contentWrapper.appendChild(descriptionElement);
+    announcementItem.appendChild(contentWrapper);
+
+    linkWrapper.appendChild(announcementItem);
+    app.appendChild(linkWrapper);
+  }
+
+  async function fetchData() {
+    try {
+      let apiData = [];
+
+      const response = await fetch(blockData.caasUrl);
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      apiData = await response.json();
+
+      if (apiData?.cards) {
+        apiData.cards.forEach((card) => {
+          const cardDate = new Date(card.cardDate);
+          if (newestCards.length < 3) {
+            newestCards.push(card);
+            newestCards.sort((a, b) => new Date(b.cardDate) - new Date(a.cardDate));
```

Comment:
Maybe we can add `size=3`  query param to the link so we fetch less data instead of all cards and then sort them ? It looks like adding this query param will return 3 newest cards and then you can sort them (but please check this I am not 100% sure)

Reply:
api doesn't return 3 cards sorted by cardDate